pkg/restclient/common.go

```diff
@@ -0,0 +1,62 @@
+package restclient
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+const baseURL = "http://localhost:24007"
+
+// RESTClient represents Glusterd2 REST Client
+type RESTClient struct {
+	baseURL  string
+	username string
+	password string
+}
+
+// NewRESTClient creates new instance of RESTClient
+func NewRESTClient(baseURL string, username string, password string) *RESTClient {
+	return &RESTClient{baseURL, username, password}
+}
+
+func httpRequest(method string, url string, respType string, body io.Reader, expectStatusCode int) (string, error) {
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return "", err
+	}
+	if respType == "xml" || respType == "json" {
+		req.Header.Set("Accept", "application/"+respType)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	resp, err1 := http.DefaultClient.Do(req)
+	if err1 != nil {
+		return "", err1
+	}
+
+	if resp.StatusCode == 404 {
```

Comment:
Idea behind this is to handle not implemented APIs smoothly in CLI, For example Same CLI can work with multiple versions of Glusterd2, if one Glusterd2(Cluster) doesn't have a API implemented then CLI can handle it. I think this check can be moved to CLI itself instead of having it in restclient.